geomagio/WebService.py

```diff
@@ -15,21 +17,16 @@ def __init__(self, factory):
 
     def __call__(self, environ, start_response):
         """Implement WSGI interface"""
+        self.query = WebServiceQuery
```

Comment:
Update WebServiceQuery.parse to return a new WebServiceQuery object, then use a local variable so concurrent requests don't overwrite each other (`query = WebServiceQuery.parse(...)`)